### ISSUE TITLE
Wordcount code of Pootle gives a wrong output sometimes

### ISSUE THREAD

**Issue by JohnnyKing94** [translate/pootle#4279]
I wanna alert you that sometime the code used by Pootle to count the words gives a wrong output. There are some phrases like this with I made experiments that Pootle calculates **44 words**, when they're ONLY **40**.

```
There is a woman in town by the name of Elsie Dolban. She is eccentric, and that has caused some of the villagers to condemn her. I'm interested in *your* opinion of her.nnSpeak to Inquisitor Roche, he has strong opinions in this matter.
```
I don't know if it's the ```*``` as special character or what... but true is, I used every tools, even the command ```wc -w``` of linux to count the words. now i'm worried about the real word count of all my strings uploaded on the db having special characters...
I alert you to take a look into it.

**Comment by iafan** [comment]
I counted the words manually (just splitting text by space and assuming `nn` here is a line break) and there are 44 words. How did you get 40?

**Comment by JohnnyKing94** [comment]
Sorry i meant 43 words... i used Microsoft Word and the "echo "THE-TEXT | wc -w" command on linux...

**Comment by iafan** [comment]
Ok. So is this `nn` thing in the middle a real line break in the string or do you have `her.nnSpeak` exactly like this and expect it to be counted as one word? Do you probably have a link to the unit in Pootle to verify against?

**Comment by JohnnyKing94** [comment]
Yes i was doing tests with the xliff format and i noted that the backslashes were missing, anyway even with ```\n\n```, still counted as 43... it doesn't change...

**Comment by unho** [comment]
Just checked on the shell:

```python
In [1]: from pootle.core.utils.wordcount import wordcount

In [2]: wordcount("There is a woman in town by the name of Elsie Dolban. She "
  ....:           "is eccentric, and that has caused some of the villagers to "
  ....:           "condemn her. I'm interested in *your* opinion of her.\n\n"
  ....:           "Speak to Inquisitor Roche, he has strong opinions in this "
  ....:           "matter.")
Out[2]: 45
```

**Comment by JohnnyKing94** [comment]
Lol, now it tells 45 :D... i have 44 with the both version "\n\n" and the "nn" on Pootle... with my other tools, Word and the linux command tells me 43... and notepad++ tells me 45... wtf is wrong? ahahahhahahahahah

**Comment by unho** [comment]
Is it ok to say Pootle provides the right count :)

It gets funnier if you try:

```python
In [3]: len(("There is a woman in town by the name of Elsie Dolban. She "
             "is eccentric, and that has caused some of the villagers to "
             "condemn her. I'm interested in *your* opinion of her.\n\n"
             "Speak to Inquisitor Roche, he has strong opinions in this "
             "matter.").split())
Out[3]: 44

```

**Comment by dwaynebailey** [comment]
@unho @JohnnyKing94 ``translate.storage.statsdb.wordcount`` is the default wordcount set by ``POOTLE_WORDCOUNT_FUNC``.  So the one being pointed to, ``pootle.core.utils.wordcount``,  is not the default wordcount function being used.

@JohnnyKing94 note that \n is not going to be seen as whitespace by either Word 2016 or to ``wc``, so you will get an under reading there.   Try ``echo "One\n\nTwo" | wc -w`` and you get a count of 1 because there is only one word there, not two, Word would have the same issue.

```py
string = """There is a woman in town by the name of Elsie Dolban. She is eccentric, and that has caused some of the villagers to condemn her. I'm interested in *your* opinion of her.\n\nSpeak to Inquisitor Roche, he has strong opinions in this matter."""

from pootle.core.utils.wordcount import wordcount as pcount
from translate.storage.statsdb import wordcount as tcount

print pcount(string)
print tcount(string)
```
This give the counts, 45 and 44 respectively.  @iafan not sure why the Pootle one counts 45.  But the default one counts 44 which by inspection looks correct.

**Comment by unho** [comment]
Merged some tests in #4287 to highlight the differences between both wordcount functions. I am closing this.
